src/utils/records.ts

```diff
@@ -263,6 +265,11 @@ export class Records {
       filterCopy.datePublished = datePublishedFilter;
     }
 
+    //if sorting by published, results must filter for published
+    if (dateSort === DateSort.PublishedAscending || dateSort === DateSort.PublishedDescending) {
+      filterCopy.published = true;
+    }
```

Comment:
There's a check for something similar but not exactly. It's a check to make sure the user doesn't set a `datePublisehd` while also having  `publisehd:false` as that would return no results. But that is self contained within the filter schema.

The filter and pagination properties are different schema objects, so there isn't any dependency link there (as it's structured currently).

Reply:
> It's a check to make sure the user doesn't set a datePublisehd while also having publisehd:false

Would it make sense to do this check in the same place? Since it feels like an integrity check, it feels strange to be part of "convertFilter".